In [1]:
!pip install numpy

In [2]:
!pip install pandas

In [3]:
import numpy as np
import pandas as pd

In [14]:
data = pd.read_csv('C:\\Users\\Mahbub\\Documents\\Welding\\Welder-health-data-collection.csv')

df = data.copy()


In [15]:
# option to display all the columns
pd.set_option('display.max_columns', None)

In [16]:
df.head()

Timestamp Serial number: (ক্রমিক সংখ্যা)      Name: (নাম)  \
0   8/3/2022 0:07:31                              1             test   
1   8/8/2022 8:30:45                             10         Md.Selim   
2  8/8/2022 18:00:57                             11        Md.Mahbub   
3  8/8/2022 18:15:54                             12        Ahmed Ali   
4  8/8/2022 18:25:00                             13  Md.Aminul Islam   

   Age: (বয়স)  Please fill your mobile number (মোবাইল নম্বর)  \
0           23                                     1578131456   
1           33                                     1710437192   
2           30                                     1999279990   
3           18                                     1903869844   
4           32                                     1722936557   

   If you have NID  then, please provide your NID number(NID নম্বর):  \
0                                       1.234568e+11                   
1                                                NaN                   
2                                                NaN                   
3                                                NaN                   
4                                                NaN                   

  Gender (লিঙ্গ)   Please specify your profession (পেশা):  \
0    Male(পুরুষ)  Full time welder (পূর্ণকালীন ওয়েল্ডার)   
1    Male(পুরুষ)  Full time welder (পূর্ণকালীন ওয়েল্ডার)   
2    Male(পুরুষ)      Multipurpose worker (বহুমুখী কর্মী)   
3    Male(পুরুষ)  Full time welder (পূর্ণকালীন ওয়েল্ডার)   
4    Male(পুরুষ)      Multipurpose worker (বহুমুখী কর্মী)   

  Which workshop do you work  for? (কর্মশালার নাম):  \
0                                              test   
1                  Mayer Dowa Engineering workshop    
2                      Saidul Engineering workshop    
3                      Saidul Engineering workshop    
4                       Sadia Engineering workshop    

  Please select your work location (কর্মস্থল):  \
0                           Meherpur(মেহেরপুর)   
1                            Rajshahi(রাজশাহী)   
2                            Rajshahi(রাজশাহী)   
3                            Rajshahi(রাজশাহী)   
4                            Rajshahi(রাজশাহী)   

   How many years have you been working?(কাজের সময়কাল)  \
0                                                  6      
1                                                 22      
2                                                 20      
3                                                  2      
4                                                 10      

  Have you received  previous training on welding? (ঝালাইয়ের উপর কোন পূর্ববর্তী প্রশিক্ষণ?)  \
0                                         Yes(হ্যাঁ)                                          
1                                             No(না)                                          
2                                             No(না)                                          
3                                             No(না)                                          
4                                             No(না)                                          

   What is your average daily work hour (দৈনিক কাজের গড় ঘন্টা):  \
0                                                  7               
1                                                 11               
2                                                 11               
3                                                 11               
4                                                 11               

  Which type of welding do you do? (ঝালাইয়ের ধরন):  \
0                                              Arc   
1                                              Arc   
2                                             Both   
3                                              Arc   
4                                              Arc   

   What is your duration span for gas welding type? ( গ্যাস \nওয়েল্ডিং ধরনের জন্য আপনার সময়কা

In [17]:
df.columns

Index(['Timestamp', 'Serial number: (ক্রমিক সংখ্যা)', 'Name: (নাম)',
       'Age: (বয়স)', 'Please fill your mobile number (মোবাইল নম্বর)',
       'If you have NID  then, please provide your NID number(NID নম্বর):',
       'Gender (লিঙ্গ)', 'Please specify your profession (পেশা):',
       'Which workshop do you work  for? (কর্মশালার নাম):',
       'Please select your work location (কর্মস্থল):',
       'How many years have you been working?(কাজের সময়কাল)',
       'Have you received  previous training on welding? (ঝালাইয়ের উপর কোন পূর্ববর্তী প্রশিক্ষণ?)',
       'What is your average daily work hour (দৈনিক কাজের গড় ঘন্টা):',
       'Which type of welding do you do? (ঝালাইয়ের ধরন):',
       'What is your duration span for gas welding type? ( গ্যাস \nওয়েল্ডিং ধরনের জন্য আপনার সময়কাল কত? ):',
       'What is your duration span for arc welding type? ( আর্ক ওয়েল্ডিং ধরনের জন্য আপনার সময়কাল কত? )',
       'What is your process of welding ?(ক্রমাগত ঝালাই  বা বিক্ষিপ্ত ঝালাই?)',
       'Do

# Breathing Issue

In [31]:
breathing_issue_data = df[['Age: (বয়স)','Gender (লিঙ্গ)','Please specify your profession (পেশা):',
    
   'How many years have you been working?(কাজের সময়কাল)',
   
   'What is your average daily work hour (দৈনিক কাজের গড় ঘন্টা):',
   'Which type of welding do you do? (ঝালাইয়ের ধরন):',
   'What is your duration span for gas welding type? ( গ্যাস \nওয়েল্ডিং ধরনের জন্য আপনার সময়কাল কত? ):',
   'What is your duration span for arc welding type? ( আর্ক ওয়েল্ডিং ধরনের জন্য আপনার সময়কাল কত? )',
   'What is your process of welding ?(ক্রমাগত ঝালাই  বা বিক্ষিপ্ত ঝালাই?)',
   'Do you use any safety tools during welding? (ঝালাইয়ের সময় নিরাপত্তা)',
   'Did you have any breathing issues before working? ( কাজ করার আগে আপনার কি শ্বাসকষ্টের সমস্যা ছিল?)',
   'Do you currently have any breathing issue now? (এখন কি শ্বাসকষ্ট হয়?)',
   'Do you have any smoking habit? (ধূমপানের অভ্যাস)',
   ]]

In [73]:
# renaming columns

column_mapping = {
    'Age: (বয়স)':'Age',
    'Gender (লিঙ্গ)':'Gender',
    'Please specify your profession (পেশা):' : 'Profession',
    'How many years have you been working?(কাজের সময়কাল)': 'Years of experience',
    'What is your average daily work hour (দৈনিক কাজের গড় ঘন্টা):': 'Average work hour(Daily)',
    'Which type of welding do you do? (ঝালাইয়ের ধরন):': 'Type of welding',
    'What is your duration span for gas welding type? ( গ্যাস \nওয়েল্ডিং ধরনের জন্য আপনার সময়কাল কত? ):': 'Duration span for gas welding',
    'What is your duration span for arc welding type? ( আর্ক ওয়েল্ডিং ধরনের জন্য আপনার সময়কাল কত? )': 'Duration span for arc welding',
    'What is your process of welding ?(ক্রমাগত ঝালাই  বা বিক্ষিপ্ত ঝালাই?)': 'Process of welding',
    'Do you use any safety tools during welding? (ঝালাইয়ের সময় নিরাপত্তা)': 'Use of safety tools',
    'Did you have any breathing issues before working? ( কাজ করার আগে আপনার কি শ্বাসকষ্টের সমস্যা ছিল?)':'Breathing issues before welding work',
    'Do you currently have any breathing issue now? (এখন কি শ্বাসকষ্ট হয়?)': 'Currently having any breathing issue',
    'Do you have any smoking habit? (ধূমপানের অভ্যাস)':'Smoking havit'
    
}


breathing_issue_data=breathing_issue_data.rename(columns=column_mapping)
breathing_issue_data

Age       Gender                               Profession  \
0     23  Male(পুরুষ)  Full time welder (পূর্ণকালীন ওয়েল্ডার)   
1     33  Male(পুরুষ)  Full time welder (পূর্ণকালীন ওয়েল্ডার)   
2     30  Male(পুরুষ)      Multipurpose worker (বহুমুখী কর্মী)   
3     18  Male(পুরুষ)  Full time welder (পূর্ণকালীন ওয়েল্ডার)   
4     32  Male(পুরুষ)      Multipurpose worker (বহুমুখী কর্মী)   
..   ...          ...                                      ...   
146   18  Male(পুরুষ)  Full time welder (পূর্ণকালীন ওয়েল্ডার)   
147   27  Male(পুরুষ)  Full time welder (পূর্ণকালীন ওয়েল্ডার)   
148   26  Male(পুরুষ)      Multipurpose worker (বহুমুখী কর্মী)   
149   38  Male(পুরুষ)      Multipurpose worker (বহুমুখী কর্মী)   
150   23  Male(পুরুষ)      Multipurpose worker (বহুমুখী কর্মী)   

     Years of experience  Average work hour(Daily) Type of welding  \
0                      6                         7             Arc   
1                     22                        11             Arc   
2                     20                        11            Both   
3                      2                        11             Arc   
4                     10                        11             Arc   
..                   ...                       ...             ...   
146                    4                         9             Arc   
147                    3                         9             Arc   
148                   10                         9             Arc   
149                   22                         9             Arc   
150                    3                        12             Arc   

     Duration span for gas welding Duration span for arc welding  \
0                              8.0                            jk   
1                              NaN                            22   
2                             10.0                            10   
3                              NaN                             2   
4                              NaN                            10   
..                             ...                           ...   
146                            NaN                             4   
147                            NaN                             3   
148                            NaN                            10   
149                            NaN                            22   
150                            NaN                             3   

    Process of welding Use of safety tools  \
0           Continuous  Glass, Hand gloves   
1              Tacking               Glass   
2              Tacking               Glass   
3              Tacking               Glass   
4              Tacking               Glass   
..                 ...                 ...   
146            Tacking               Glass   
147         Continuous               Glass   
148         Continuous               Glass   
149         Continuous               Glass   
150            Tacking               Glass   

    Breathing issues before welding work Currently having any breathing issue  \
0                             Yes(হ্যাঁ)                               No(না)   
1                                 No(না)                               No(না)   
2                                 No(না)                               No(না)   
3                                 No(না)                               No(না)   
4                                 No(না)                               No(না)   
..                                   ...                                  ...   
146                               No(না)                               No(না)   
147                               No(না)                               No(না)   
148                               No(না)                               No(না)   
149                               No(না)                               No(না)   
150                               No(না)                               No(না)   

    Smoking havit  
0      Y

In [74]:
breathing_issue_data.head()

Age       Gender                               Profession  \
0   23  Male(পুরুষ)  Full time welder (পূর্ণকালীন ওয়েল্ডার)   
1   33  Male(পুরুষ)  Full time welder (পূর্ণকালীন ওয়েল্ডার)   
2   30  Male(পুরুষ)      Multipurpose worker (বহুমুখী কর্মী)   
3   18  Male(পুরুষ)  Full time welder (পূর্ণকালীন ওয়েল্ডার)   
4   32  Male(পুরুষ)      Multipurpose worker (বহুমুখী কর্মী)   

   Years of experience  Average work hour(Daily) Type of welding  \
0                    6                         7             Arc   
1                   22                        11             Arc   
2                   20                        11            Both   
3                    2                        11             Arc   
4                   10                        11             Arc   

   Duration span for gas welding Duration span for arc welding  \
0                            8.0                            jk   
1                            NaN                            22   
2                           10.0                            10   
3                            NaN                             2   
4                            NaN                            10   

  Process of welding Use of safety tools Breathing issues before welding work  \
0         Continuous  Glass, Hand gloves                           Yes(হ্যাঁ)   
1            Tacking               Glass                               No(না)   
2            Tacking               Glass                               No(না)   
3            Tacking               Glass                               No(না)   
4            Tacking               Glass                               No(না)   

  Currently having any breathing issue Smoking havit  
0                               No(না)    Yes(হ্যাঁ)  
1                               No(না)    Yes(হ্যাঁ)  
2                               No(না)    Yes(হ্যাঁ)  
3                               No(না)        No(না)  
4                               No(না)    Yes(হ্যাঁ)

In [75]:
breathing_issue_data.columns

Index(['Age', 'Gender', 'Profession', 'Years of experience',
       'Average work hour(Daily)', 'Type of welding',
       'Duration span for gas welding', 'Duration span for arc welding',
       'Process of welding', 'Use of safety tools',
       'Breathing issues before welding work',
       'Currently having any breathing issue', 'Smoking havit'],
      dtype='object')

### Number of responses for before breathing issue?

In [77]:
count = breathing_issue_data['Breathing issues before welding work'].value_counts()
count

No(না)        146
Yes(হ্যাঁ)      5
Name: Breathing issues before welding work, dtype: int64

### Percentage of welders who didn't have breathing issue before.

In [78]:
count = breathing_issue_data['Breathing issues before welding work'].value_counts()['No(না)']
total_count = breathing_issue_data['Breathing issues before welding work'].count()
percentage = (count / total_count) * 100
had_no_issues = percentage.round(2)
had_no_issues

96.69

##### So, 96.69% people did not have breathing issues.

### Percentage of welders who had breathing issue before welding

In [79]:
had_breathing_issue = (100-percentage)
had_breathing_issue.round(2)

3.31

##### So, 3.31 % people had breathing issues before starting welding

### Number of responses for after welding breathing issue?

In [81]:
count = breathing_issue_data['Currently having any breathing issue'].value_counts()
count

No(না)        131
Yes(হ্যাঁ)     20
Name: Currently having any breathing issue, dtype: int64

### Percentage of welders who do not have breathing issue now

In [82]:
count = breathing_issue_data['Currently having any breathing issue'].value_counts()['No(না)']
total_count = breathing_issue_data['Currently having any breathing issue'].count()
percentage = (count / total_count) * 100
having_no_issues = percentage.round(2)
having_no_issues

86.75

##### 86.75% people do not have breathing issue now.

### Percentage of welders who have breathing issue now

In [84]:
count = breathing_issue_data['Currently having any breathing issue'].value_counts()['Yes(হ্যাঁ)']
total_count = breathing_issue_data['Currently having any breathing issue'].count()
percentage = (count / total_count) * 100
percentage.round(2)
having_issues = percentage.round(2)
having_issues

13.25

##### 13.25% people has breathing issue now.

### How much breathing issues increases?

In [85]:
increased_issues = had_no_issues - having_no_issues
increased_issues.round(2)

9.94

##### Breathing issue has been increased by 9.94 %


### Percentage of people who are having breathing issue do welding and also smokes

In [104]:
current_breath_issues_and_smokes = breathing_issue_data[(breathing_issue_data['Currently having any breathing issue'] == 'Yes(হ্যাঁ)') 
                                               &(breathing_issue_data['Smoking havit'] == 'Yes(হ্যাঁ)')].reset_index()[['Currently having any breathing issue','Smoking havit']]

In [115]:
len(current_breath_issues_and_smokes)

9

In [116]:
len(breathing_issue_data)

151

In [117]:
# percentage

(len(current_breath_issues_and_smokes) / len(breathing_issue_data)) * 100

5.960264900662252

##### So, 5.96 % people have breathing issues who do both welding and smoking

### Percentage of people who are having breathing issue do welding but do not smoke

In [119]:
current_breath_issues_but_do_not_smoke = breathing_issue_data[(breathing_issue_data['Currently having any breathing issue'] == 'Yes(হ্যাঁ)') 
    &(breathing_issue_data['Smoking havit'] == 'No(না)')].reset_index()[['Currently having any breathing issue','Smoking havit']]

In [120]:
current_breath_issues_but_do_not_smoke

Currently having any breathing issue Smoking havit
0                            Yes(হ্যাঁ)        No(না)
1                            Yes(হ্যাঁ)        No(না)
2                            Yes(হ্যাঁ)        No(না)
3                            Yes(হ্যাঁ)        No(না)
4                            Yes(হ্যাঁ)        No(না)
5                            Yes(হ্যাঁ)        No(না)
6                            Yes(হ্যাঁ)        No(না)
7                            Yes(হ্যাঁ)        No(না)
8                            Yes(হ্যাঁ)        No(না)
9                            Yes(হ্যাঁ)        No(না)
10                           Yes(হ্যাঁ)        No(না)

In [121]:
len(current_breath_issues_but_do_not_smoke)

11

In [122]:
len(breathing_issue_data)

151

In [123]:
# percentage

(len(current_breath_issues_but_do_not_smoke) / len(breathing_issue_data)) * 100

7.28476821192053

##### So, 7.28% people have breathing issues who do welding but don't smoke